protocol/contracts/governance/Templar.sol

```diff
@@ -0,0 +1,89 @@
+//SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+
+/**
+ * @title an NFT representing membership in the temple community.
+ *
+ * It exists as on chain proof that an account owns a discord id.
+ */
+contract Templar is ERC721, Ownable, AccessControl {
+
+    bytes32 public constant CAN_ASSIGN = keccak256("CAN_ASSIGN");
+
+    string public baseUri = "https://discordapp.com/users/";
+
+     // Mapping from discordId ID to max temple role
+    mapping(uint256 => string) public templeRole;
+
+    constructor() ERC721("Templar", "TEMPLAR") {
+    }
+
+    function supportsInterface(bytes4 interfaceId) public view override(ERC721, AccessControl) returns (bool) {
+        return (
+            ERC721.supportsInterface(interfaceId) || 
+            AccessControl.supportsInterface(interfaceId)
+        );
+    }
+
+    /**
+     *  Update the base URI for tokens.
+     */
+    function setBaseUri(string calldata _baseUri) external onlyOwner {
+        baseUri = _baseUri;
```

Comment:
add event

Reply:
It's not clear to me what value an event would add here. How do you see it being used?

Reply:
Best practice for all protocol set style events to emit events such they can be responded too off-chain. slither type tools will suggest so too

In this case, hypothetically, it would allow something like a metadata sync to be triggered for all nfts in the collection.

Reply:
OK - I'll add it. I though this was only so that tools like metamask coujld show a link.